frontend/public/components/modals/roll-back-rc-modal.jsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+import { createModalLauncher, ModalTitle, ModalBody, ModalSubmitFooter } from '../factory/modal';
+import { withHandlePromise } from '../utils';
+import { DeploymentConfigModel } from '../../models';
+import { k8sCreate, k8sUpdate } from '../../module/k8s';
+
+const BaseRollBackRCModal = withHandlePromise((props) => {
+  const { t } = useTranslation();
+  const dcName = props.resource.metadata.ownerReferences[0].name;
+  const [changeScaleSettings, setChangeScaleSettings] = React.useState(false);
+  const [changeStrategy, setChangeStrategy] = React.useState(false);
+  const [changeTriggers, setChangeTriggers] = React.useState(false);
+
+  const submit = (e) => {
+    e.preventDefault();
+
+    // put together a new rollback request
+    const req = {
+      kind: 'DeploymentConfigRollback',
+      apiVersion: 'apps.openshift.io/v1',
+      name: dcName,
+      latest: true,
+      force: true,
+      spec: {
+        from: {},
+        revision: parseInt(
+          props.resource?.metadata?.annotations?.['openshift.io/deployment-config.latest-version'],
+          10,
+        ),
+        includeTemplate: true,
+        includeReplicationMeta: changeScaleSettings,
+        includeStrategy: changeStrategy,
+        includeTriggers: changeTriggers,
+      },
+    };
+    const opts = {
+      name: dcName,
+      ns: props.resource.metadata.namespace,
+      path: 'rollback',
+    };
+
+    return props.handlePromise(
+      // create the deployment config rollback
+      k8sCreate(DeploymentConfigModel, req, opts).then((updatedDC) => {
+        // update the deployment config based on the one returned by the rollback
+        return k8sUpdate(DeploymentConfigModel, updatedDC);
+      }),
+      props.close,
+    );
+  };
+
+  return (
+    <form onSubmit={submit} name="form" className="modal-content">
+      <ModalTitle>{t('modal~Roll Back')}</ModalTitle>
```

Comment:
Should "Rollback" be one word? (FWIW, that's what's used in the both the OpenShift and k8s documentation.)

Reply:
Was going by the 3.x images. I'll find out for sure.

Reply:
We could use `Rollback` but would we still, then, use `when rolling back` ?

Reply:
Updated. Also updated screen shots above